src/test/fuzz/policy_estimator.cpp

```diff
@@ -48,7 +48,7 @@ FUZZ_TARGET(policy_estimator, .init = initialize_policy_estimator)
                 const auto tx_info = NewMempoolTransactionInfo(entry.GetSharedTx(), entry.GetFee(),
                                                                entry.GetTxSize(), entry.GetHeight(),
                                                                /* m_mempool_limit_bypassed */ false,
-                                                               /* m_submitted_in_package */ false,
+                                                               /* m_submitted_in_package */ fuzzed_data_provider.ConsumeBool(),
```

Comment:
Hm, as far as I know there's no easy way... I once found a bug like that where a deterministic random generator was used, and the unit test was asserting the exact output of an algorithm given a fixed but random input. On Windows the result was different because there the compiler had chosen a different evaluation order.

Reply:
Fixed in https://github.com/bitcoin/bitcoin/commit/fab46cc454f94b33dda6bc68a7234b439310f358 thank you @martinus

Reply:
@maflcko I just looked through the code a bit to find other usages of `fuzzed_data_provider` that depend on the evaluation order, and there are quite a few of them. E.g.

* https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/addrman.cpp#L272
* https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/addrman.cpp#L286-L290
* https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/banman.cpp#L79
* https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/bloom_filter.cpp#L22-L26
* https://github.com/martinus/bitcoin/blob/master/src/test/fuzz/buffered_file.cpp#L28
* ...

It would be great if it were possible to find these automatically, but as far as I know that doesn't exist anywhere. Maybe with a custom clang-tidy plugin for specific classes, (random generator and fuzzer)?

Reply:
I created #29043

Reply:
Thanks @martinus Will review